lib/daru/vector.rb

```diff
@@ -238,9 +238,15 @@ def [](*indexes)
           indexes = indexes.map { |e| named_index_for(e) }
         end
 
-        Daru::Vector.new(
-          indexes.map { |loc| @data[@index[loc]] }, 
-          name: @name, index: indexes, dtype: @dtype)
+        begin
+          Daru::Vector.new(
+            indexes.map { |loc| @data[@index[loc]] },
+            name: @name, index: indexes, dtype: @dtype)
+        rescue NoMethodError
+          raise IndexError, "Specified index #{pos.inspect} does not exist."
+        rescue TypeError
+          raise IndexError, "One or more of specified indices #{pos.inspect} do not exist."
```

Comment:
@v0dro I am confused here. `v[[:x, :y]]` does not return a TypeError and nor does `v[[[:x, :y]]]` if `:x` and `:y` are valid indexes. So, should I treat them as wrong input types?